pkgs/top-level/python-packages.nix

```diff
@@ -12319,6 +12319,10 @@ self: super: with self; {
 
   torch = callPackage ../development/python-modules/torch {
     cudaSupport = pkgs.config.cudaSupport or false;
+    magma =
+      if pkgs.config.cudaSupport or false
+      then pkgs.magma-cuda-static
+      else pkgs.magma;
```

Comment:
Should we move this down to package definition?
With the current approach,

- 👍🏻 `(import <nixpkgs> { config.cudaSupport = true; }).torch` correctly uses static magma
- 👎🏻 `(import <nixpkgs> { }).torch.override { cudaSupport = true; }` uses shared magma

Lowering the defaults to package-level, the second expression would also use static magma